apps/src/panels/PanelsLabView.tsx

```diff
@@ -0,0 +1,60 @@
+// Panels
+//
+// This is a React client for a panels level.  Note that this is
+// only used for levels that use Lab2.
+
+import React, {useCallback} from 'react';
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+import {
+  sendSuccessReport,
+  navigateToNextLevel,
+} from '@cdo/apps/code-studio/progressRedux';
+import {PanelsLevelData} from './types';
+import PanelsView from './PanelsView';
+import useWindowSize from '../util/hooks/useWindowSize';
+
+const appName = 'panels';
+
+const PanelsLabView: React.FunctionComponent = () => {
+  const dispatch = useAppDispatch();
+
+  const levelData = useAppSelector(
+    state => state.lab.levelProperties?.levelData
+  );
+  const currentAppName = useAppSelector(
+    state => state.lab.levelProperties?.appName
+  );
+
+  const onContinue = useCallback(
+    (nextUrl?: string) => {
+      if (nextUrl) {
+        // This is a short-term solution for the Music Lab progression in incubation.  We will not attempt
+        // to send a success report for a level that uses this feature.
+        window.location.href = nextUrl;
+      } else {
+        dispatch(sendSuccessReport(appName));
+        dispatch(navigateToNextLevel());
+      }
+    },
+    [dispatch]
+  );
+
+  const panels = (levelData as PanelsLevelData | undefined)?.panels;
```

Comment:
nit: Just for the context, is there a way to don't use `as` here? (What makes us use as here?)

Reply:
Yeah, this is because the `levelData` attribute that the lab2 framework provides is just a base type that specific labs extend (like here with `PanelsLevelData`). That being said, it doesn't feel great to have to cast it everywhere it's used, so might be better to move these types up into Lab2 and create union types out of all of them.